Add H3 package and the library required for it

mdales opened the pull request:
This is my first attempt to submit a package to Opam - apologies for the inevitable mistakes, please do point me in the right direction!

This PR adds two packages:

* An OCaml wrapper for the H3 hex tile based geospatial library from Uber
* A meta package that builds and installs in opam the H3 library itself

The meta package is used for two reasons:

* Support for H3 in standard repositories is patchy
* There are differences in where they install the library (e.g., homebrew puts the headers in an h3 subdirectory, ubuntu does not).

avsm commented:
@kit-ty-kate I'm less convinced by that; `h3` is a widely used tile format, where h1/h2 are not widely accepted short forms for http versions.  Should we really rename an upstream package because other people have picked short names close to this one?

mdales commented:
I've fixed the following:

* Renamed the C library wrapper to remove conf
* Fixed silly ls -la in opam file
* Fixed description in opam file

@avsm I've not yet done the vendor recommendation, as the Yaml example didn't quite match what I'd want to do here (still using the source build system).

@kit-ty-kate I didn't make the name H3, that's the name used in the geospatial community for this product. If it's a show stopper then we could rename it, but this is the name I'd expect any geospatial would be expecting.

mdales commented:
@mseri thanks. Between this and @avsm's comments I'll close this PR and make a new one once I've understood vendoring C libraries.
